src/vsr/replica.zig

```diff
@@ -6251,15 +6253,40 @@ pub fn ReplicaType(
                         .op_max = range.op_max,
                     }),
                 );
-                return;
+
+                if (range.op_max > self.op_checkpoint()) {
+                    // If there's a header break after checkpoint, wait until it is repaired.
+                    // Otherwise, concurrently repair headers before checkpoint, while trying
+                    // to commit everything after the checkpoint.
+                    return;
+                }
             }
-            assert(self.valid_hash_chain_between(self.op_repair_min(), self.op));
+            assert(self.valid_hash_chain_between(
+                @min(self.op_checkpoint() + 1, self.op),
+                self.op,
+            ));
 
             if (self.journal.dirty.count > 0) {
                 // Request and repair any dirty or faulty prepares.
-                self.repair_prepares();
-            } else if (self.client_replies.faulty.findFirstSet()) |slot| {
-                // After we have all prepares, repair replies.
```

Comment:
My understanding is that there's actually no reason why we mush repair prepares before proceeding to repair replies, but I am not totally sure.

Reply:
Yep, no reason.